server/task/src/main/java/org/apache/james/task/TaskWithId.java

```diff
@@ -0,0 +1,52 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.task;
+
+import java.util.Objects;
+
+public class TaskWithId {
+    private final TaskId id;
+    private final Task task;
+
+    public TaskWithId(TaskId id, Task task) {
+        this.id = id;
+        this.task = task;
+    }
+
+    public TaskId getId() {
+        return id;
+    }
+
+    public Task getTask() {
+        return task;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
This POJO is not tested. Please EqualVerify this...

Reply:
I don't see how it could pass the EqualVerify test, has the encapsulated Task does not verify it...